File: frontend/src/services/orderService.js
import axios from 'axios';

const API_URL = '/api/orders';

const orderService = {
  createOrder: async (orderData) => {
    const response = await axios.post(API_URL, orderData);
    return response.data;
  },

  getUserOrders: async () => {
    const response = await axios.get(`${API_URL}/my-orders`);
    return response.data;
  },

  getOrderById: async (id) => {
    const response = await axios.get(`${API_URL}/${id}`);
    return response.data;
  },

  getAllOrders: async () => {
    const response = await axios.get(API_URL);
    return response.data;
  },

  updateOrderStatus: async (id, status) => {
    const response = await axios.put(`${API_URL}/${id}/status`, { status });
    return response.data;
  }
};

export default orderService;

File: backend/src/main/java/com/ecommerce/service/OrderService.java
package com.ecommerce.service;

import com.ecommerce.model.Order;
import com.ecommerce.model.User;
import com.ecommerce.repository.OrderRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

/**
 * Service class for Order management
 * Handles order CRUD operations and business logic
 */
@Service
@RequiredArgsConstructor
@Transactional
public class OrderService {

    private final OrderRepository orderRepository;

    /**
     * Create a new order
     * @param order the order to create
     * @return the created order
     */
    public Order createOrder(Order order) {
        return orderRepository.save(order);
    }

    /**
     * Find order by ID
     * @param id the order ID
     * @return the order
     */
    @Transactional(readOnly = true)
    public Order findById(Long id) {
        return orderRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Order not found with id: " + id));
    }

    /**
     * Get all orders
     * @return list of all orders
     */
    @Transactional(readOnly = true)
    public List<Order> getAllOrders() {
        return orderRepository.findAll();
    }

    /**
     * Get orders by user
     * @param user the user
     * @return list of user's orders
     */
    @Transactional(readOnly = true)
    public List<Order> getOrdersByUser(User user) {
        return orderRepository.findByUser(user);
    }

    /**
     * Get orders by user ID
     * @param userId the user ID
     * @return list of user's orders
     */
    @Transactional(readOnly = true)
    public List<Order> getOrdersByUserId(Long userId) {
        return orderRepository.findByUserId(userId);
    }

    /**
     * Update order status
     * @param id the order ID
     * @param status the new status
     * @return the updated order
     */
    public Order updateOrderStatus(Long id, Order.OrderStatus status) {
        Order order = findById(id);
        order.setStatus(status);
        return orderRepository.save(order);
    }

    /**
     * Cancel order
     * @param id the order ID
     * @return the cancelled order
     */
    public Order cancelOrder(Long id) {
        return updateOrderStatus(id, Order.OrderStatus.CANCELLED);
    }
}
